projects/testing/src/main.c

```diff
@@ -0,0 +1,38 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#include <stdint.h>
+
+typedef struct Counters {
+  uint8_t counter_a, counter_b;
+} Counters_t;
+
+void run(SoftTimerId timer_id, void *ctx) {
```

Comment:
for module internal functions we prefix with `prv_` and generally make them static, also more descriptive names are appreciated e.g. `prv_timer_callback` or `prv_periodic_count` or similar